api/src/main/resources/workbench.yaml

```diff
@@ -1592,6 +1623,31 @@ definitions:
         description: >
           Updated workspace etag after the share request has been applied.
 
+  CloneWorkspaceRequest:
+    type: object
+    required:
+     - workspace
+    properties:
+      workspace:
+        $ref: "#/definitions/Workspace"
+        description: >
+          Workspace metadata to be applied to the cloned workspace upon creation. The following
+          fields are accepted:
+            - name (required)
+            - namespace (required)
+            - researchPurpose (required)
```

Comment:
Does this contemplate the research purpose as ... a string? Or a full structured purpose as we collect when creating a workspace from scratch? Assuming the latter, this works great (will be good to reuse the structure for research purpose  if we can, i.e. the `ResearchPurpose` object from the this swagger file).

Reply:
The full structure. These refer to fields on the `Workspace` object, which includes a structured `researchPurpose` field. Reworded the above document to make this clearer.

Note: only a subset of the fields on the research purpose will actually be settable, e.g. `approvalTime` will be ignored, following the same rules `CreateWorkspace` should (but by my reading - actually doesn't currently, which I will fix).